plugins/AuthUI/src/androidMain/kotlin/io/github/temk0/supabase/authui/ui/GoogleButton.kt

```diff
@@ -0,0 +1,73 @@
+package io.github.temk0.supabase.authui.ui
+
+import android.app.Activity
+import androidx.activity.compose.rememberLauncherForActivityResult
+import androidx.activity.result.IntentSenderRequest
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.remember
+import androidx.compose.ui.platform.LocalContext
+import com.google.android.gms.auth.api.identity.Identity
+import com.google.android.gms.common.api.ApiException
+import com.google.android.gms.common.api.CommonStatusCodes
+import io.github.temk0.supabase.authui.AuthUI
+import io.github.temk0.supabase.authui.AuthUIImpl
+import io.github.temk0.supabase.authui.GoogleLoginConfig
+import io.github.temk0.supabase.authui.getSignInRequest
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.tasks.await
+
+@Composable
+actual fun AuthUI.rememberLoginWithGoogle(
+    onResult: (NativeSignInResult) -> Unit,
+    fallback: () -> Unit
+): NativeSignInState {
+    this as AuthUIImpl
+
+    val state = remember { NativeSignInState() }
+    val context = LocalContext.current
+
+    // init signInRequest options
+    if (config.loginConfig == null || config.loginConfig !is GoogleLoginConfig) {
+        fallback.invoke()
+        state.reset()
+        return state
+    }
```

Comment:
The fallback should also only be invoked in the LaunchedEffect block as it's getting called on every composition